mm2src/crypto/src/hw_client.rs

```diff
@@ -124,26 +124,37 @@ impl HwClient {
     ) -> MmResult<TrezorClient, HwProcessingError<Processor::Error>> {
         use common::custom_futures::timeout::TimeoutError;
         use common::executor::Timer;
+        use trezor::transport::ConnectableDeviceWrapper;
 
-        async fn try_to_connect() -> HwResult<Option<TrezorClient>> {
-            let mut devices = trezor::transport::usb::find_devices()?;
+        async fn try_to_connect<C>() -> HwResult<Option<TrezorClient>>
+        where
+            C: ConnectableDeviceWrapper + 'static,
+        {
+            let mut devices = C::find_devices().await?;
             if devices.is_empty() {
                 return Ok(None);
             }
             if devices.len() != 1 {
                 return MmError::err(HwError::CannotChooseDevice { count: devices.len() });
             }
             let device = devices.remove(0);
-            let transport = device.connect()?;
+            let transport = device.connect().await?;
             let trezor = TrezorClient::from_transport(transport);
             Ok(Some(trezor))
         }
 
         let fut = async move {
             loop {
-                if let Some(trezor) = try_to_connect().await? {
+                if let Some(trezor) = try_to_connect::<trezor::transport::usb::UsbAvailableDevice>().await? {
                     return Ok(trezor);
                 }
+
+                #[cfg(feature = "trezor-udp")]
+                // try also to connect to emulator over UDP
+                if let Some(trezor) = try_to_connect::<trezor::transport::udp::UdpAvailableDevice>().await? {
+                    return Ok(trezor);
+                }
```

Comment:
I guess it should be enough to check whether if mm2 running on debug mode or not? Adding a feature for just this piece of code seems an overkill to me.

Reply:
My idea for adding udp as a feature was that normally we would use those tests for trezor with a physical device (that is why I marked them 'ignored' so you need to have a device to run them) and only who wishes to build and use emulator would enable this feature.
(Later we will add ci tests which include emulator auto build and start)

Reply:
> /// Tool to run withdraw directly with trezor device or emulator (no rpc version, added for easier debugging)
/// run cargo test with '--ignored' option

But we have many other tests marked as ignored too, so this test will end up with lots of other tests running and most of them probably will fail. Here we should actually use the feature flag e.g., `cargo test --features enable-device-tests`.

Reply:
I see that `run-device-tests` feature is added, please confirm that this is fixed @dimxy

Reply:
Yes I added a feature "run-device-tests" (to invoke trezor tests, if needed)